test/e2e/resourceinterpreter_test.go

```diff
@@ -271,6 +280,23 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
 			})
 		})
 	})
+
+	ginkgo.Context("InterpreterOperation InterpretDependency testing", func() {
```

Comment:
We can set the relevant cm only for operation InterpretDependency, such as:
```diff
@@ -30,7 +30,6 @@ import (
 var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
        var policyNamespace, policyName string
        var workloadNamespace, workloadName string
-       var configMapNamespace, configMapName string
        var workload *workloadv1alpha1.Workload
        var policy *policyv1alpha1.PropagationPolicy
        var cm *corev1.ConfigMap
@@ -40,10 +39,7 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
                policyName = workloadNamePrefix + rand.String(RandomStrLength)
                workloadNamespace = testNamespace
                workloadName = policyName
-               configMapNamespace = testNamespace
-               configMapName = configMapNamePrefix + rand.String(RandomStrLength)
-
-               workload = testhelper.NewWorkload(workloadNamespace, workloadName, configMapName)
+               workload = testhelper.NewWorkload(workloadNamespace, workloadName)
                policy = testhelper.NewPropagationPolicy(policyNamespace, policyName, []policyv1alpha1.ResourceSelector{
                        {
                                APIVersion: workload.APIVersion,
@@ -55,19 +51,14 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
                                ClusterNames: framework.ClusterNames(),
                        },
                })
-               // configmaps should be propagated automatically.
-               policy.Spec.PropagateDeps = true
-               cm = testhelper.NewConfigMap(configMapNamespace, configMapName, map[string]string{"RUN_ENV": "test"})
        })

        ginkgo.JustBeforeEach(func() {
                framework.CreatePropagationPolicy(karmadaClient, policy)
                framework.CreateWorkload(dynamicClient, workload)
-               framework.CreateConfigMap(kubeClient, cm)
                ginkgo.DeferCleanup(func() {
                        framework.RemoveWorkload(dynamicClient, workload.Namespace, workload.Name)
                        framework.RemovePropagationPolicy(karmadaClient, policy.Namespace, policy.Name)
-                       framework.RemoveConfigMap(kubeClient, configMapNamespace, configMapName)
                })
        })

@@ -282,6 +273,28 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
        })

        ginkgo.Context("InterpreterOperation InterpretDependency testing", func() {
+               var configMapNamespace, configMapName string
+
+               ginkgo.BeforeEach(func() {
+                       configMapNamespace = testNamespace
+                       configMapName = configMapNamePrefix + rand.String(RandomStrLength)
+
+                       workload.Spec.Template.Spec.Containers[0].EnvFrom = []corev1.EnvFromSource{
+                               {ConfigMapRef: &corev1.ConfigMapEnvSource{
+                                       LocalObjectReference: corev1.LocalObjectReference{Name: configMapName},
+                                       Optional:             pointer.Bool(true),
+                               }}}
+
+                       // configmaps should be propagated automatically.
+                       policy.Spec.PropagateDeps = true
+                       cm = testhelper.NewConfigMap(configMapNamespace, configMapName, map[string]string{"RUN_ENV": "test"})
+
+                       framework.CreateConfigMap(kubeClient, cm)
+                       ginkgo.DeferCleanup(func() {
+                               framework.RemoveConfigMap(kubeClient, configMapNamespace, configMapName)
+                       })
+               })
+
                ginkgo.It("InterpretDependency testing", func() {
```

Reply:
You're right, but is it really necessary to tweak? maybe it will be used in the future for additional test cases, I think.

Reply:
You're right, but is it really necessary to tweak? maybe it will be used in the future for additional test cases, I think.

Reply:
I'm sorry I don't get it.

Reply:
Sorry, I typo the wrong words, what I want to say is:

For other operations, propagating comfigmap is unnecessary because it consumes extra resources and may affect our analysis of e2e when it goes wrong.